nsz/NszDecompressor.py

```diff
@@ -77,71 +49,57 @@ def __decompress(filePath, outputDir = None, write = True, raiseVerificationExce
 			elif not write:
 				Print.info('[EXISTS]     {0}'.format(nspf._path))
 			continue
-
 		newFileName = nspf._path[0:-1] + 'a'
 		if write:
 			f = newNsp.add(newFileName, nspf.size)
 			start = f.tell()
 		blockID = 0
 		nspf.seek(0)
-		
 		header = nspf.read(ncaHeaderSize)
 		magic = nspf.read(8)
 		if not magic == b'NCZSECTN':
 			raise ValueError("No NCZSECTN found! Is this really a .ncz file?")
 		sectionCount = nspf.readInt64()
-		sections = []
-		for i in range(sectionCount):
-			sections.append(Header.Section(nspf))
-
+		sections = [Header.Section(nspf) for _ in range(sectionCount)]
 		pos = nspf.tell()
 		blockMagic = nspf.read(8)
 		nspf.seek(pos)
 		useBlockCompression = blockMagic == b'NCZBLOCK'
-		
 		blockSize = -1
 		if useBlockCompression:
 			BlockHeader = Header.Block(nspf)
 			blockDecompressorReader = BlockDecompressorReader.BlockDecompressorReader(nspf, BlockHeader)
 		pos = nspf.tell()
-
-		dctx = zstandard.ZstdDecompressor()
 		if not useBlockCompression:
-			decompressor = dctx.stream_reader(nspf)
-
-		hash = hashlib.sha256()
+			decompressor = ZstdDecompressor().stream_reader(nspf)
+		hash = sha256()
 		with tqdm(total=nspf.size, unit_scale=True, unit="B") as bar:
 			if write:
 				f.write(header)
 			bar.update(len(header))
 			hash.update(header)
-			
+
 			for s in sections:
 				i = s.offset
-				
 				crypto = aes128.AESCTR(s.cryptoKey, s.cryptoCounter)
 				end = s.offset + s.size
-				
 				while i < end:
 					crypto.seek(i)
 					chunkSz = 0x10000 if end - i > 0x10000 else end - i
 					if useBlockCompression:
 						inputChunk = blockDecompressorReader.read(chunkSz)
 					else:
 						inputChunk = decompressor.read(chunkSz)
-					
+					bar.update(chunkSz/8)
```

Comment:
This seems to break the progress bar when decompressing. I don't really understand why it should be progressing with chunkSz/8

Reply:
Oh, sorry about that. I was just testing the tdqm update, will fix it soon on next commit to my repo.